src/tests/HttpMessageTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+use AWS\CRT\HTTP\Message;
+use AWS\CRT\HTTP\Headers;
+use AWS\CRT\HTTP\Request;
+use AWS\CRT\HTTP\Response;
+
+require_once('common.inc');
+
+final class HttpMessageTest extends CrtTestCase {
+    public function testHeaders() {
+        $headers = new Headers();
+        $this->assertEquals(0, $headers->count());
+    }
+
+    public function testHeadersMarshalling() {
+        $headers_array = [
+            "host" => "s3.amazonaws.com",
+            "test" => "this is a test header value"
+        ];
+        $headers = new Headers($headers_array);
+        $this->assertEquals(2, $headers->count());
```

Comment:
AssertSame > assertEquals if you aren't comparing different types

Reply:
kind of a nitpick though, this works too

Reply:
== vs === in PHP matters